Remotion/Validation/UnitTests/Validators/NotEmptyCollectionValidatorTest.cs

```diff
@@ -0,0 +1,219 @@
+    {
+      var propertyValidatorContext = CreatePropertyValidatorContext(new GenericCollection<string> { "someValue" });
+      var validator = new NotEmptyCollectionValidator(new InvariantValidationMessage("Fake Message"));
+
+      var validationFailures = validator.Validate(propertyValidatorContext).ToArray();
+
+      Assert.That(validationFailures, Is.Empty);
+    }
+
+    [Test]
+    public void Validate_WithPropertyValueIsNonEmptyReadOnlyCollection_ReturnsNoValidationFailures ()
+    {
+      var propertyValidatorContext = CreatePropertyValidatorContext(new ReadOnlyCollection<string> { "someValue" });
+      var validator = new NotEmptyCollectionValidator(new InvariantValidationMessage("Fake Message"));
+
+      var validationFailures = validator.Validate(propertyValidatorContext).ToArray();
+
+      Assert.That(validationFailures, Is.Empty);
+    }
+
+    [Test]
+    public void Validate_WithPropertyValueIsEmptyString_ReturnsNoValidationFailures ()
+    {
+      var propertyValidatorContext = CreatePropertyValidatorContext(string.Empty);
+      var validator = new NotEmptyCollectionValidator(new InvariantValidationMessage("Fake Message"));
+
+      var validationFailures = validator.Validate(propertyValidatorContext).ToArray();
+
+      Assert.That(validationFailures, Is.Empty);
+    }
+
+    [Test]
+    public void Validate_WithPropertyValueIsEmptyBinary_ReturnSingleValidationFailure ()
+    {
+      var propertyValidatorContext = CreatePropertyValidatorContext(Array.Empty<byte>());
+      var validator = new NotEmptyCollectionValidator(new InvariantValidationMessage("Fake Message"));
+
+      var validationFailures = validator.Validate(propertyValidatorContext).ToArray();
+
+      Assert.That(validationFailures.Length, Is.EqualTo(1));
+      Assert.That(validationFailures[0].ErrorMessage, Is.EqualTo("The value must not be empty."));
+    }
+
+    [Test]
+    public void Validate_WithPropertyValueIsObject_ReturnsSingleValidationFailure ()
+    {
+      var propertyValidatorContext = CreatePropertyValidatorContext(new object());
+      var validator = new NotEmptyCollectionValidator(new InvariantValidationMessage("Fake Message"));
+
+      var validationFailures = validator.Validate(propertyValidatorContext).ToArray();
+
+      Assert.That(validationFailures, Is.Empty);
+    }
+
+    private class Enumerable<T> : IEnumerable<T>
+    {
+      private readonly List<T> _list = new();
+
+      public IEnumerator<T> GetEnumerator () => _list.GetEnumerator();
+
+      IEnumerator IEnumerable.GetEnumerator () => GetEnumerator();
+    }
+
+    private class SimpleCollection : ICollection
+    {
+      private readonly List<object> _list = new();
+
+      public IEnumerator GetEnumerator () => _list.GetEnumerator();
+
+      public void CopyTo (Array array, int index)
+      {
+        var objectArray = array.Cast<object>().ToArray();
+        _list.CopyTo(objectArray, index);
+      }
```

Comment:
Oh, there's a bug here. Since we're not using the API in the tests, it's not relevant, tough we should either fix make it through an exception instead. Otherwise, someone might find this and copy it to production.

The bug: the CopyTo method copies the list into the array. Since we create a new array from the parameter, the list is copied into the new array which is then discarded. the array from the parameter remains unchanged.